admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js

```diff
@@ -78,6 +78,9 @@ function (ich,Marionette,_,$,Q,ModalSource,EmptyView,Service,Status,wreqr,Utils,
             }
             data.available = this.model.get('available');
             data.name = this.model.get('name');
+            if (!data.name && data.currentConfiguration && data.currentConfiguration.id) {
```

Comment:
That's not what's happening here though. The model has all the data, just not in the right place, and should instead map more accurately to the template. I don't think any of this logic belongs in the view, but rewriting the model seemed out of scope for this. I could still do that, there just might be a lot more side-effect changes.